doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -1748,7 +1660,7 @@ Tactic notations allow to customize the syntax of tactics. They have the followi
    This information is particularly relevant for notations of tacticals.
    Levels 0 to 5 are available (default is 5).
 
-   .. cmd:: Print Grammar tactic
+   .. cmd:: Print Grammar @ident
```

Comment:
It can be more than the three above. For instance, it can also be `pattern`. In principle, all defined grammars can be printed, but the `Print Grammar` is missing some tables iirc.

Reply:
@herbelin `pattern` was in the list of the "three above". What did you mean to say? In any case, only these three were documented. Where can we find the list of all defined grammars?

Reply:
Oups... Rectification, the only missing in the list is `vernac`.

The code is in `Metasyntax.pr_grammar` and the only non hard-wired grammar is `tactic` which the `ltac_plugin` registers. Other plugins are in principle allowed to register grammars too.

This is currently not active, but it would be feasible to allow printing the grammars for `sort`, `universe`, ... if ever useful.

Reply:
Here lie dragons:
- the documentation presents an edited grammar that is equivalent to but that doesn't match this too closely--some nonterminal names are different, some productions have been combined in the doc. and other things.  For example, `Print Grammar vernac` shows entries for `vernac_control`, `syntax`, `gallina` and `gallina_ext` that we don't show in the documentation.
- productions are shown with a different syntax than in the doc, e.g. `  | IDENT "Section"; Prim.identref` versus

![image](https://user-images.githubusercontent.com/1253341/80269595-f648b380-8665-11ea-856e-7e741b2ec156.png)

We don't use `IDENT <string>` and `Prim.identref` is just `ident` in the doc.

Furthermore, the output is not very user friendly.  It doesn't show Tactic Notations.  Perhaps it is useful to developers for debugging Coq.  Not sure if it's worth keeping.

A related issue is that the nonterminals mentioned in parser error messages don't match what we're showing in the documentation.  That's probably worth fixing at some point.

I will fix the writeup for `Print Grammar`.,

Reply:
Yes, once we've finished updating the doc in the manual, it would be great to change the parser as much as possible to match closely the manual, so that understanding parsing errors becomes easier and can be done with the help of the manual.